src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,488 @@
+                File.ReadAllText(Krb5ConfigFile).Contains(TestConfiguration.Realm);
+            if (!_isKrbInstalled)
+            {
+                password = Guid.NewGuid().ToString("N");
+                int exitCode = RunSetupScript(string.Format(ScriptInstallArgs, password));
+                if (exitCode != 0)
+                {
+                    if (exitCode != 2)
+                    {
+                        Dispose();
+                    }
+
+                    Assert.True(false, "KDC setup failure");
+                }
+            }
+            else
+            {
+                password = TestConfiguration.DefaultPassword;
+            }
+        }
+
+        public void Dispose()
+        {
+            if (!_isKrbInstalled)
+            {
+                RunSetupScript(ScriptUninstallArgs);
+            }
+        }
+
+        // checks for avilability of Kerberos related infrastructure
+        // on the host. Returns true available, false otherwise
+        public bool CheckAndClearCredentials(ITestOutputHelper output)
+        {
+            // Clear the credentials
+            var startInfo = new ProcessStartInfo(KDestroyCmd);
+            startInfo.UseShellExecute = true;
+            startInfo.CreateNoWindow = true;
+            startInfo.Arguments = "-A";
+            using (Process clearCreds = Process.Start(startInfo))
+            {
+                clearCreds.WaitForExit();
+                output.WriteLine("kdestroy returned {0}", clearCreds.ExitCode);
+                return (clearCreds.ExitCode == 0);
+            }
+        }
+
+        private static int RunSetupScript(string args = null)
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo();
+
+            // since ProcessStartInfo does not support Verb, we use sudo as
+            // the program to be run
+            startInfo.FileName = SudoCommand;
+            startInfo.Arguments = string.Format("bash {0} {1}", ScriptName, args);
+            using (Process kdcSetup = Process.Start(startInfo))
+            {
+                kdcSetup.WaitForExit();
+                return kdcSetup.ExitCode;
+            }
+        }
+    }
+
+    public class KerberosTest : IDisposable, IClassFixture<KDCSetup>
+    {
+        private readonly byte[] _firstMessage = Encoding.UTF8.GetBytes("Sample First Message");
+        private readonly byte[] _secondMessage = Encoding.UTF8.GetBytes("Sample Second Message");
+        private readonly bool _isKrbAvailable; // tests are no-op if kerberos is not available on the host machine
+        private readonly KDCSetup _fixture;
+        private readonly ITestOutputHelper _output;
+
+        public KerberosTest(KDCSetup fixture, ITestOutputHelper output)
+        {
+            _fixture = fixture;
+            _output = output;
+            _isKrbAvailable = _fixture.CheckAndClearCredentials(_output);
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux)]
```

Comment:
Rather than putting [PlatformSpecific(PlatformID.Linux)] on each test, you should be able to put it on the class.